### PR TITLE
ddl: fix reorg cannot handle divide 0 error

### LINKED ISSUE #50053: should report error when divide 0 occurs when adding a generated index with default sql mode
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
> create table t1(a int);
> insert into t1 values(0);
> alter table t1 add index i((100/a));
```

### 2. What did you expect to see? (Required)

In mysql8.0

```
mysql> alter table t1 add index i((100/a));
ERROR 1365 (22012): Division by 0
```

### 3. What did you see instead (Required)

```
mysql> alter table t1 add index i((100/a));
Query OK, 0 rows affected (0.76 sec)
```

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #50053

### What changed and how does it work?

fix reorg cannot handle divide 0 error

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix reorg cannot handle divide 0 error
```

### CODE REVIEW

File: pkg/ddl/backfilling_scheduler.go [hunk 1]
```diff
@@ -163,6 +163,7 @@ func initSessCtx(
 	if err := setSessCtxLocation(sessCtx, tzLocation); err != nil {
 		return errors.Trace(err)
 	}
+	sessCtx.GetSessionVars().StmtCtx.InReorg = true
```
Comment (tangenta) [inline pkg/ddl/backfilling_scheduler.go:166]: We need to restore this value in `restoreSessCtx` before putting back to session pool.

Comment (lcwangchao) [reply pkg/ddl/backfilling_scheduler.go:166]: done

**Review by tangenta — APPROVED** [review]

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**